src/sql-parser/src/ident.rs

```diff
@@ -64,8 +76,32 @@ macro_rules! ident {
             }
         }
 
+        #[allow(dead_code)]
```

Comment:
I'm guessing this is dead code in the case when the length is invalid, then we panic and never check values?

Reply:
Interestingly, the compiler considers it dead even in the happy path case, i.e., when using `ident!("foobar")`.

Here is a small [playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=93efd5e145ff6fd91cc4db2fa667c768) to experiment with. Surprisingly the two check functions require suppressing the warning, whereas `equals` doesn't. Perhaps the compiler diagnostics are not entirely accurate when it comes to `const` usage?

Reply:
> Perhaps the compiler diagnostics are not entirely accurate when it comes to const usage?

This is almost definitely it, thanks for the Playground example too!